code/game/objects/items/hand_item.dm

```diff
@@ -62,13 +62,26 @@
 /obj/item/slapper/parry
 	desc = "This is how real men win fights."
 	force = 5
+	flags = DROPDEL | ABSTRACT | NODROP
 	attack_verb = list("slapped", "backhanded", "smacked", "discombobulated")
 	table_smacks_left = 10 //Much more smackitude
 
 /obj/item/slapper/parry/Initialize(mapload)
 	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS, _parry_cooldown = (1 / 3) SECONDS) //75% uptime
+	if(isliving(loc))
+		var/mob/owner = loc
+		RegisterSignal(owner, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(drop_slapper), override = TRUE)
+		RegisterSignal(owner, COMSIG_MOB_WEAPON_APPEARS, PROC_REF(drop_slapper), override = TRUE)
 	return ..()
 
+/obj/item/slapper/parry/proc/drop_slapper()
+	SIGNAL_HANDLER
+	if(isliving(loc))
+		var/mob/owner = loc
+		UnregisterSignal(owner, COMSIG_MOB_WILLINGLY_DROP)
+		UnregisterSignal(owner, COMSIG_MOB_WEAPON_APPEARS)
+	qdel(src)
```

Comment:
won't drop del handle this?

Reply:
Does it complain runtime about deleting something multiple times?

Reply:
There was no runtimes during my testing, though i didnt try qdel, just dropped it with Q
Why you're asking? I see no loops here

Reply:
Was just interested, thanks for testing!